src/Calendar.js

```diff
@@ -688,11 +688,12 @@ class Calendar extends React.Component {
     );
   }
 
-  handleRangeChange = (date) => {
+  handleRangeChange = (date, nextView) => {
     let { onRangeChange } = this.props
+    let View = nextView ? VIEWS[nextView] : this.getView();
```

Comment:
I used the VIEWS array in my original PR but it was rejected for breaking custom views, so I think we need a different solution here.